code/headers/usart_connection.hpp

```diff
@@ -1,52 +1,90 @@
-/// @file
-/// @brief      base class for a usart implementation hardware, software and
-/// mock
-/// @author     Patrick Dekker
 #pragma once
+/**
+ * @file usart_connection.hpp
+ * @author Patrick Dekker
+ * @brief base class for a usart implementation.
+ * @version 0.1
+ * @date 2019-05-24
+ *
+ * @copyright Copyright (c) 2019
+ *
+ */
 
 #include <cstdint>
 
 namespace r2d2::usart {
-    /// @brief  base class for usart implementation
+    /**
+     * @brief Base class for the usart implementation
+     *
+     */
     class usart_connection_c {
     public:
-        /// @brief operator << writes byte to usart
-        /// @param byte byte to send
+        /**
+         * @brief operator << writes a byte with the usart
+         *
+         * @param byte to send
+         * @return usart_connection_c&
+         */
         usart_connection_c &operator<<(uint8_t byte);
 
-        /// @brief operator << writes string to usart
-        /// @param c c-style string to send
+        /**
+         * @brief operator << writes a string with the usart
+         *
+         * @param c c-style string to send
+         * @return usart_connection_c&
+         */
         usart_connection_c &operator<<(const char *c);
```

Comment:
Would a `ostream` overload make more sense?

Reply:
Made it a istream as well